File: src/test/java/edu/kit/iti/formal/smv/TypeTest.java
package edu.kit.iti.formal.smv;

import edu.kit.iti.formal.smv.ast.GroundDataType;
import edu.kit.iti.formal.smv.ast.SMVType;
import junit.framework.Assert;
import org.junit.Test;

import java.util.Arrays;

/**
 * @author Alexander Weigl
 * @version 1 (15.12.16)
 */
public class TypeTest {
    @Test
    public void testOutput() {
        SMVType.SMVTypeWithWidth u10 = new SMVType.SMVTypeWithWidth(GroundDataType.UNSIGNED_WORD, 10);
        SMVType.SMVTypeWithWidth s0 = new SMVType.SMVTypeWithWidth(GroundDataType.SIGNED_WORD, 0);
        SMVType b = new SMVType();
        SMVType.EnumType e = new SMVType.EnumType(Arrays.asList("a", "b", "c"));

        Assert.assertEquals("unsigned word[10]", u10.toString());
        Assert.assertEquals("boolean", b.toString());
        Assert.assertEquals("signed word[0]", s0.toString());
        Assert.assertEquals("{a, b, c}", e.toString());


    }

}

File: smv/src/test/kotlin/edu/kit/iti/formal/smv/TypeTest.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.smv

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

/**
 * @author Alexander Weigl
 * @version 1 (15.12.16)
 */
class TypeTest {
    @Test
    fun testOutput() {
        val u10 = SMVWordType(false, 10)
        val s0 = SMVWordType(true, 0)
        val b = SMVTypes.BOOLEAN
        val e = EnumType(Arrays.asList("a", "b", "c"))

        Assertions.assertEquals("unsigned word[10]", u10.repr())
        Assertions.assertEquals("boolean", b.repr())
        Assertions.assertEquals("signed word[0]", s0.repr())
        Assertions.assertEquals("{a, b, c}", e.repr())
    }
}